### PR TITLE
Gra 1198 enrollment keys

### PR DESCRIPTION
**Pull request by 0xdcarns**
Do not merge till post 0.18.1
Testing steps:

- [x] Ensure you can create a enrollment keys only with admin privileges
- `curl -X POST -H "Authorization: Bearer <admin JWT>" -d '{"expiration":0,"uses_remaining":10,"networks":["networktoJoin1","network2"],"unlimited":false,"tags":[]}' https://<mydomain.com>/api/v1/enrollment-keys`
  `curl -H "Authorization: Bearer <admin JWT>" https://mydomain.com/api/v1/enrollment-keys | jq`
uses_remaining = how many uses are on it
expiration is set with unix time
unlimited indicates unlimited use one of those must be set, otherwise should fail

- 

- [x] Ensure you can get enrollment keys only with admin privileges

- `curl -H "Authorization: Bearer <admin JWT>" https://<mydomain.com>/api/v1/enrollment-keys | jq`
- [x] Ensure you can delete an enrollment key only with admin privilege
- `curl -X DELETE -H "Authorization: Bearer <admin JWT>" https://<mydomain.com>/api/v1/enrollment-keys/<keyValue>`

- [x] Check PR https://github.com/gravitl/netclient/pull/223 for netclient usability

### CODE REVIEW

File: controllers/enrollmentkeys.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	"github.com/gravitl/netmaker/logger"
+	"github.com/gravitl/netmaker/logic"
+	"github.com/gravitl/netmaker/logic/hostactions"
+	"github.com/gravitl/netmaker/models"
+	"github.com/gravitl/netmaker/mq"
+	"github.com/gravitl/netmaker/servercfg"
+)
+
+func enrollmentKeyHandlers(r *mux.Router) {
+	r.HandleFunc("/api/v1/enrollment-keys", logic.SecurityCheck(true, http.HandlerFunc(createEnrollmentKey))).Methods(http.MethodPost)
```
[the remark this answers is not in the document]
Comment (0xdcarns) [reply controllers/enrollmentkeys.go:19]: I'm doing it for this one

**Review by Aceix — CHANGES_REQUESTED** [review]

File: models/enrollment_key.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+type APIEnrollmentKey struct {
+	Expiration    int64    `json:"expiration"`
+	UsesRemaining int      `json:"uses_remaining"`
+	Networks      []string `json:"networks"`
+	Unlimited     bool     `json:"unlimited"`
+	Tags          []string `json:"tags"`
+}
+
+// EnrollmentKey.IsValid - checks if the key is still valid to use
+func (k *EnrollmentKey) IsValid() bool {
```
[the remark this answers is not in the document]
Comment (0xdcarns) [reply models/enrollment_key.go:38]: that's not correct

**Review by Aceix — APPROVED** [review]

**Review by mattkasun — APPROVED** [review]

**Review by afeiszli — CHANGES_REQUESTED** [review]
Should be able to use more than one enrollment key after joining a server, to join additional networks.

**Review by theguy951357 — APPROVED** [review]
checkmarks on pr pass
critical pr checklist 100%
pass

**Pull request merged**